packages/rocketchat-lib/server/models/Rooms.js

```diff
@@ -561,9 +561,9 @@ class ModelRooms extends RocketChat.models._Base {
 		return this.update(query, update);
 	}
 
-	incMsgCountAndSetLastMessageById(_id, inc, lastMessageTimestamp, lastMessage) {
+	incMsgCountAndSetLastMessageById(room, token, inc, lastMessageTimestamp, lastMessage) {
```

Comment:
If I'm not wrong, your implementation here is not related to `msgCount` and `LastMessage` stuff, but is related to `room.metrics` informations. Because of this, please create a new method in the model to store these data, reverting the changes inside `incMsgCountAndSetLastMessageById`.